### ISSUE TITLE
Moving directory reimports image assets and recalculates all metadata

### ISSUE THREAD

**Issue by CitrusWire** [godotengine/godot#92083]
### Tested versions

4.2.2 - reproducable

### System information

Windows 7

### Issue description

Moving a directory of already imported assets (jpgs in this case) re-imports them at great computational expense.

I don't know what the import process entails, but given the insane temporary resource usage (12GB of RAM for this use-case), I expect part of it is analysing the images themselves to gather metadata, which makes sense. However, when moving a directory, Godot should only need to change relative file locations in the metadata; not do anything computationally expensive like recalculate all of the imagery metadata.

### Steps to reproduce

1) Create a new project
2) add a directory of several hundred MB of pictures to the project. Ideally some of them should be 30+MB in size.
3) Create a subfolder.
3) In godot, drag the directory of images into the subfolder

### Minimal reproduction project (MRP)

N/A

**Comment by huwpascoe** [comment]
This always happened for me whenever files are moved, I assumed that's how it's supposed to work? If not then it'd be convenient to have that fixed.

**Comment by CitrusWire** [comment]
> This always happened for me whenever files are moved, I assumed that's how it's supposed to work? If not then it'd be convenient to have that fixed.

Yes, this may be how it works, however it's low-hanging fruit for optimisation. Well, low-hanging on a conceptual level at least, implementation I have no idea about. I guess this is why they flagged it as "performance".

**Comment by AThousandShips** [comment]
It's far from simple to handle this necessarily, depending on how the test is done it isn't guaranteed it can be assumed the file hasn't changed and needs reimport, though it could be reasonable to assume when moved in the editor

**Comment by CitrusWire** [comment]
> though it could be reasonable to assume when moved in the editor

That's the use-case I'm referring to here.

> it isn't guaranteed it can be assumed the file hasn't changed and needs reimport [if you're doing the move outside of godot editor]

That one seems fairly simply to solve too: Keep a hash of the file contents. Upon re-import, rehash the file. If it's the same, it hasn't changed and no need to re-do the computationally expensive stuff.

**Comment by AThousandShips** [comment]
> [if you're doing the move outside of godot editor]

Please do not inject your own assumptions into quotes by me, that's not what I meant at all

But as I said, it *might* be trivial, but needs investigation, hence the `needs testing`

**Comment by CitrusWire** [comment]
> Please do not inject your own assumptions into quotes by me, that's not what I meant at all

I was clarifying what I meant, that's why it's in square brackets.

**Comment by AThousandShips** [comment]
Then do not present it as a quote please, because you are not quoting me, you are putting words in my mouth :) So just say what *you* mean instead of making claims about what *I* mean, thank you.

However as I said it depends on the test and when and how it's done, it depends on what we assume even when moving files in the editor

**Comment by KoBeWi** [comment]
This is another case of #23107

**Comment by precup** [comment]
@AThousandShips I can confirm this is still happening on v4.5.dev.custom_build [19bb18716], Linux Mint X11. Could debatably be a duplicate of the issue mentioned above.
